sphinx/ext/intersphinx.py

```diff
@@ -273,22 +274,43 @@ def load_mappings(app: Sphinx) -> None:
         # Duplicate values in different inventories will shadow each
         # other; which one will override which can vary between builds
         # since they are specified using an unordered dict.  To make
-        # it more consistent, we sort the named inventories and then
-        # add the unnamed inventories last.  This means that the
-        # unnamed inventories will shadow the named ones but the named
-        # ones can still be accessed when the name is specified.
-        named_vals = []
-        unnamed_vals = []
+        # it more consistent, we sort the named inventories by their
+        # name and then add the unnamed inventories last.  This means
+        # that the unnamed inventories will shadow the named ones but
+        # the named ones can still be accessed when the name is specified.
+        #
+        # When we encounter a named inventory that already exists,
+        # this means that we had two entries for the same inventory,
+        # but with different URIs (e.g., the remote documentation was
+        # updated).  In particular, the newest URI is inserted in the
+        # cache *after* the one that existed from a previous build.
+        #
+        # Example: assume that we use A to generate the inventory of "foo",
+        # so that we have
+        #
+        #   intersphinx_cache = {A: ('foo', timeout, D1)}
+        #
+        # If we rebuild the project, the cache becomes
+        #
+        #   intersphinx_cache = {A: ('foo', ..., D1), B: ('foo', ..., D2)}
```

Comment:
In these two examples, can I check that `A` and `B` are the `name` of external projects, and `'foo'` represents a URL?

If so, I'd suggest an adjustment to make that clearer:
```suggestion
        #   intersphinx_cache = {PROJ_A: ('http://example.org/proj_a', timeout, D1)}
        #
        # If we rebuild the project, the cache becomes
        #
        #   intersphinx_cache = {PROJ_A: ('http://example.org/proj_a', ..., D1), PROJ_B: ('https://example.com/proj_b', ..., D2)}
```

Also: what do `D1` and `D2` represent?

Reply:
> In these two examples, can I check that A and B are the name of external projects, and 'foo' represents a URL?

Actually, no. The cache is build differently. The indexation key is the URL and "foo" is the real name of the project. I can however change A/B to URI1 and URI2.